gdal/ogr/ogrsf_frmts/mssqlspatial/ogrmssqlspatialtablelayer.cpp

```diff
@@ -2259,6 +2266,15 @@ OGRErr OGRMSSQLSpatialTableLayer::ICreateFeature( OGRFeature *poFeature )
 
         return OGRERR_FAILURE;
     }
+    else if(pszFIDColumn != nullptr && bIsIdentityFid)
+    {
+        // fetch new ID and set it into the feature
+        if (oStatement.Fetch())
```

Comment:
just wondering what will happen in the case CreateFeature() was called with a FID != OGRNullFID. Will oStatement.Fetch() return false, and so the original FID will be kept ?

Reply:
I thought that I had a reason at the time; however, since the OUTPUT clause is only added based on all three conditions

```
        /* prepend VALUES section */
        if (nFID == OGRNullFID && pszFIDColumn != nullptr && bIsIdentityFid)
            oStatement.Appendf(") OUTPUT INSERTED.[%s] VALUES (", GetFIDColumn());
        else
            oStatement.Appendf( ") VALUES (" );

```
it does stand to reason that the same full condition check should be considered when fetching.

I will submit an update with all three conditions.